tests/multi_ptr/multi_ptr_access_members.h

```diff
@@ -43,42 +68,33 @@ class run_access_members_tests {
     auto queue = sycl_cts::util::get_cts_object::queue();
     constexpr int val_to_init = 42;
     T value = user_def_types::get_init_value_helper<T>(val_to_init);
-    // Variables that will be used to check that access members returns correct
-    // data type
-    bool dereference_return_type_is_correct = false;
-    bool dereference_op_return_type_is_correct = false;
-    bool get_return_type_is_correct = false;
-    bool get_raw_return_type_is_correct = false;
-    bool get_decorated_return_type_is_correct = false;
-    // Variables that will be used to check that access members returns correct
-    // value
-    T dereference_ret_value;
-    T dereference_op_ret_value;
-    T get_member_ret_value;
-    T get_raw_member_ret_value;
+
+    // Variable that contains all variables that will be used to verify test
+    // result
+    detail::test_result<T> test_result;
 
     sycl::range r = sycl::range(1);
     {
-      sycl::buffer<T> dereference_ret_value_buffer(&dereference_ret_value,
-                                                   sycl::range(1));
-      sycl::buffer<T> dereference_op_ret_value_buffer(&dereference_op_ret_value,
-                                                      sycl::range(1));
-      sycl::buffer<T> get_member_ret_value_buffer(&get_member_ret_value,
-                                                  sycl::range(1));
-      sycl::buffer<T> get_raw_member_ret_value_buffer(&get_raw_member_ret_value,
-                                                      sycl::range(1));
+      sycl::buffer<T> dereference_ret_value_buffer(
+          &test_result.dereference_ret_value, sycl::range(1));
```

Comment:
I suggest we create a single buffer for the structure object instead of creating buffers for each member. Otherwise assembling all these values into a single structure doesn't make much sense.

Reply:
oh, that's is a good idea, I've done this in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/346/commits/8b67da5742fe39063c9d4d9416acf4d1cfeb0d46